openstudiocore/src/energyplus/ForwardTranslator/ForwardTranslateCoilHeatingDesuperheater.cpp

```diff
@@ -104,13 +101,36 @@ boost::optional<IdfObject> ForwardTranslator::translateCoilHeatingDesuperheater(
   boost::optional<ModelObject> heatingSource = modelObject.heatingSource();
 
   if ( heatingSource ) {
-      boost::optional<IdfObject> _heatingSource = translateAndMapModelObject(heatingSource.get());
+    if (boost::optional<IdfObject> _heatingSource = translateAndMapModelObject(*heatingSource)) {
```

Comment:
I thought that we decided against assigning optional in conditional tests.  It might be fine, but just the possibility that it isn't scares me off

Reply:
I never knew this was frown upon until now (and that there was a possibility it wouldn't work?). I'll remember that going forward then, and modify this line and the equivalent in the other file.

Reply:
94ea298a0be22770faebccbcfe5eb988c8ddb0d2